adapta/utils/_common.py

```diff
@@ -188,7 +190,34 @@ def map_column_names(
     return dataframe
 
 
-def downcast_dataframe(dataframe: DataFrame, columns: Optional[List[str]] = None) -> DataFrame:
+def map_column_names_polars(
+    dataframe: polars.DataFrame,
+    column_map: Dict[str, str],
+    default_values: Optional[Dict[str, Union[str, int, float]]] = None,
+    drop_missing: bool = True,
+) -> polars.DataFrame:
+    """
+    Maps a dataframe from one nomenclature to another. Original dataframe is not mutated.
+
+    :param dataframe: Dataframe to be mapped.
+    :param column_map: A dictionary mapping old column names to new.
+    :param default_values: If a column is not present in the dataframe
+    a default value mapping can be given, by mapping a column name it a value.
+    :param drop_missing: A boolean value to control if columns should be
+    dropped if the columns are present in the dataframe but not the column_map.
+    """
+    default_values = default_values or {}
+    # Only columns in the map are mapped
+    kept_columns = list(set(column_map.keys()) & set(dataframe.columns)) if drop_missing else dataframe.columns
+    column_map = {k: v for k, v in column_map.items() if k in kept_columns}
+    dataframe = dataframe.select(kept_columns).rename(column_map)
```

Comment:
add this to the return line, no reason for this allocation

Reply:
done

Reply:
it was actually needed, added back